tracer/src/Datadog.Trace/ClrProfiler/AutoInstrumentation/MongoDb/MongoDbTags.cs

```diff
@@ -31,4 +32,34 @@ internal partial class MongoDbTags : InstrumentationTags
         [Tag(Trace.Tags.OutPort)]
         public string Port { get; set; }
     }
+
+    internal partial class MongoDbV1Tags : MongoDbTags
+    {
+        private string _peerServiceOverride = null;
+
+        // Use a private setter for setting the "peer.service" tag so we avoid
+        // accidentally setting the value ourselves and instead calculate the
+        // value from predefined precursor attributes.
+        // However, this can still be set from ITags.SetTag so the user can
+        // customize the value if they wish.
+        [Tag(Trace.Tags.PeerService)]
+        public string PeerService
+        {
+            get => _peerServiceOverride ?? DbName ?? Host;
+            private set => _peerServiceOverride = value;
+        }
+
+        [Tag(Trace.Tags.PeerServiceSource)]
+        public string PeerServiceSource
+        {
+            get
+            {
+                return _peerServiceOverride is not null
+                        ? "peer.service"
+                        : DbName is not null
+                            ? "db.instance"
+                            : "network.destination.name";
+            }
+        }
```

Comment:
I wonder if we should have a couple of simple unit tests around this behaviour, mostly to make sure we don't accidentally break it in the future, as I don't think this is covered by integration tests yet either?

Reply:
The only one that's not covered right now is the "peer.service" manual setting. In the MongoDb 2.2 snapshots, you can see peer.service getting set to the DbName and also the Host name (mongo)

Reply:
But I think you're right, small unit tests would be good for these tags and their logic. I'll go ahead and add some

Reply:
Added in 1bd86c282ca6461b69d7d01a9fd27b602b601982